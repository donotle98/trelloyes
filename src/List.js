import React from "react";
import Cards from "./Card";

export default function List(props) {
  return (
    <section className="List">
      <header>
        <h2>{props.header}</h2>
      </header>
      <div className="List-cards">
        {props.cards.map((card) => (
          <Cards
            key={card.id}
            id={card.id}
            title={card.title}
            content={card.content}
            onClickDelete={props.onClickDelete}
          />
        ))}
        <button
          type="button"
          className="List-add-button"
          onClick={() => props.onClickAdd(props.id)}
        >
          + Add Random Card
        </button>
      </div>
    </section>
  );
}
List.defaultProps = {
  onClickAdd: () => {},
};
