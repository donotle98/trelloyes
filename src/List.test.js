import React from "react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";
import List from "./List";
import store from "./store";

describe("here is the test for list", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<List cards={[]} />, div);

    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the UI as expected", () => {
    const tree = TestRenderer.create(
      <List
        key={"1"}
        header={"First list"}
        cards={["a", "b", "e", "f", "g", "j", "l", "m"]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
