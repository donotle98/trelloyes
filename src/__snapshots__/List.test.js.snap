// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`here is the test for list renders the UI as expected 1`] = `
<section
  className="List"
>
  <header>
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <button
      class="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
